tests/PHPCR/Tests/Util/QOM/Sql2ScannerTest.php

```diff
@@ -124,6 +124,57 @@ public function testSQLEscapedStrings2()
         $this->expectTokensFromScanner($scanner, $expected);
     }
 
+    public function testSquareBrackets()
+    {
+        $sql = 'WHERE ISSAMENODE(file, ["/home node"])';
+
+        $scanner = new Sql2Scanner($sql);
+        $expected = [
+            'WHERE',
+            'ISSAMENODE',
+            '(',
+            'file',
+            ',',
+            '["/home node"]',
```

Comment:
Since we didn't have tests before. I couldn't tell that I changed the logic. Would have probably made more sense to create the tests first. But I that shouldn't be too hard to implement. The question still remains then though, what about invalid strings inside those brackets. Should they be validated?